ambry-quota/src/main/java/com/github/ambry/quota/QuotaUtils.java

```diff
@@ -39,4 +43,32 @@ public static boolean isRequestResourceQuotaManaged(RestRequest restRequest) {
         Operations.ACCOUNTS) || requestPath.matchesOperation(Operations.STATS_REPORT) || requestPath.matchesOperation(
         Operations.ACCOUNTS_CONTAINERS));
   }
+
+  public static boolean isReadRequest(RestRequest restRequest) {
+    switch (restRequest.getRestMethod()) {
+      case GET:
+      case OPTIONS:
+      case HEAD:
+        return true;
+      default:
+        return false;
+    }
+  }
+
+  public static String getQuotaResourceId(RestRequest restRequest) {
```

Comment:
Is it possible to have a collision on account and container names? Should we also return the resource type?